Sources/FluentKit/Migration/Migrator.swift

```diff
@@ -174,10 +214,18 @@ public struct Migrator {
             }
         }
     }
-    
+
+
     private func database(_ id: DatabaseID?) -> Database {
         self.databaseFactory(id)
     }
+
+    private func run(on database: DatabaseID? = nil, _ query: @escaping (DatabaseID?) -> EventLoopFuture<Void>) -> EventLoopFuture<Void> {
```

Comment:
I'm a bit confused about what this method is doing. Perhaps what is currently `Migrator` should become `DatabaseMigrator` and handle migrating a single database? `Migrator` could then be a utility that handles running one `DatabaseMigrator` per database. We should be able to do something like that without breaking API and I think some separation of concern would make this internal code more easily understandable.